### PR TITLE
add *.csproj support to examples/gitserver/hooks/detect-language

### PR DESCRIPTION
**Pull request by jim-minter**
@bparees, I accidentally missed this off #11896.

### CODE REVIEW

File: examples/gitserver/hooks/detect-language [hunk 1]
```diff
@@ -14,6 +14,11 @@ set -o pipefail
 function has {
   [[ -n $(git ls-tree --full-name --name-only HEAD ${@:1}) ]]
 }
+function hasglob {
+  git ls-tree --full-name --name-only HEAD | while read NAME; do
+    [[ "$NAME" == $1 ]] && return 0
```
Comment (bparees) [inline examples/gitserver/hooks/detect-language]: isn't this going to compare "*.csproj" with "somefile.csproj" and fail?  or is there some bash behavior here i'm not aware of?

Comment (jim-minter) [inline examples/gitserver/hooks/detect-language]: The latter, I believe.  See `bash(1)` under `[[ expression ]]`: When  the  ==  and != operators are used, the string to the right of the operator is considered a pattern and matched according to the rules described below under Pattern Matching, as if the extglob shell option were enabled.

```bash
$ mkdir /tmp/empty && cd /tmp/empty
$ >bar.csproj
$ [[ "foo.csproj" == *.csproj ]] && echo yes
yes
$ [[ *.csproj == "foo.csproj" ]] && echo yes
$ [[ "foo.csproj" == "*.csproj" ]] && echo yes
$ [ "foo.csproj" == *.csproj ] && echo yes
$
```

```

Comment (bparees) [inline examples/gitserver/hooks/detect-language]: yeah i tested w/o the [[ ]] which is why it didn't work for me.  looks good, thanks.

**Comment by bparees** [comment]
[merge]

**Pull request merged**